### PR TITLE
Updated Test Details page form UI

### LINKED ISSUE #9120: Sample Test Details page ui updation
**Describe the bug**
Sample Test Details page ui updation

**To Reproduce**

1. Go to 'Sample Test button on sidebar'
2. Click on 'click any Sample Details button from thre card'
3. See error

**Expected behavior**
UI that's clean and easy to read

**Screenshots**
Image

i would like to change like this
Image

**Desktop (please complete the following information):**

- OS: [windows]
- Browser [chrome]
- Version [130.0.6723.117]

### PR DESCRIPTION
**Pull request by modamaan**
## Proposed Changes

- Fixes #9120 

older
Image

updated
Screenshot 2024-11-16 025300

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Release Notes

- **New Features**
  - Enhanced UI for the Sample Details component, improving layout and visual clarity.
  - Introduced a new Badge component with multiple styling variants for better status representation.

- **Bug Fixes**
  - Improved error handling and loading states in the Sample Details component.

- **Documentation**
  - Updated documentation to reflect changes in the Sample Details and Badge components.

### CODE REVIEW

File: src/components/ui/badge.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+export interface BadgeProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    VariantProps<typeof badgeVariants> {}
+
+function Badge({ className, variant, ...props }: BadgeProps) {
+  return (
+    <div className={cn(badgeVariants({ variant }), className)} {...props} />
+  );
+}
```
[the remark this answers is not in the document]
Comment (modamaan) [reply src/components/ui/badge.tsx]: added

**Comment by nihal467** [comment]
@modamaan check through the coderabbit suggestion , fix the issues with the PR, and once the PR is ready for review Tag me

File: src/components/ui/card.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  HTMLDivElement,
+  React.HTMLAttributes<HTMLDivElement>
+>(({ className, ...props }, ref) => (
+  <div
+    ref={ref}
+    className={cn("text-sm text-gray-500 dark:text-gray-400", className)}
+    {...props}
+  />
+));
+CardDescription.displayName = "CardDescription";
```
[the remark this answers is not in the document]
Comment (modamaan) [reply src/components/ui/card.tsx]: resolved

**Comment by modamaan** [comment]
@nihal467 can you check

**Comment by modamaan** [comment]
@nihal467 merge conflict solved

**Comment by Jacobjeevan** [comment]
@modamaan Test 4 is failing.

**Comment by modamaan** [comment]
> @modamaan Test 4 is failing.

 @Jacobjeevan  Test 4 is failing solved

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]
Let's replace the ButtonV2 on page with button from ui 👍

**Comment by modamaan** [comment]
> Let's replace the ButtonV2 on page with button from ui 👍

@Jacobjeevan  replace buttonV2 and add shadcn Button from ui is that ok

**Comment by Jacobjeevan** [comment]
> > Let's replace the ButtonV2 on page with button from ui 👍
> 
> @Jacobjeevan replace buttonV2 and add shadcn Button from ui is that ok

Yes, that is what I meant. Also, let's not use custom background color there. 

https://github.com/ohcnetwork/care_fe/pull/9157/files adds the primary variety; copy that over and use that 👍 

Note: Also you don't need to request review on top of tagging me 😅 I'll get the notification emails since I've already reviewed the PR.

**Comment by modamaan** [comment]
@Jacobjeevan  Button updated
Screenshot 2024-12-02 005623

File: src/components/Patient/SampleDetails.tsx [hunk 3]
```diff
@@ -305,169 +308,224 @@ export const SampleDetails = ({ id }: DetailRoute) => {
...
+                <Badge id="diff_diagnosis" variant={"secondary"}>
+                  {" "}
+                  {sampleDetails?.diff_diagnosis}
+                </Badge>
+              </div>
+            )}
+            {sampleDetails?.etiology_identified && (
+              <div className="flex justify-between items-center gap-2 p-3 bg-gray-100 rounded-lg">
+                <div className="font-medium text-sm">
+                  {t("Etiology identified")}:{" "}
```
Comment (Jacobjeevan) [inline src/components/Patient/SampleDetails.tsx]: Compare and fix the translations; they are based on key value pair, not the application directly translating the terms.

This should be etiology_identified for example.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/Patient/SampleDetails.tsx (hunk 3 above)
Comment (modamaan) [reply src/components/Patient/SampleDetails.tsx]: @Jacobjeevan resolved

**Review by Jacobjeevan — APPROVED** [review]

**Comment by nihal467** [comment]
@modamaan 
image

- [ ] multiple translation are missing, do verify there are translation for all status and other variables

**Comment by modamaan** [comment]
@nihal467 
Screenshot 2024-12-03 064026
The production code's status result is already like this


Screenshot 2024-12-03 064532
Translation compared with production code and corrected

**Comment by nihal467** [comment]
LGTM

File: src/components/Patient/SampleDetails.tsx [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
-import { navigate } from "raviger";
+import { Link, navigate } from "raviger";
 import { useTranslation } from "react-i18next";
 
 import Card from "@/CAREUI/display/Card";
 
-import ButtonV2 from "@/components/Common/ButtonV2";
+import { Badge } from "@/components/ui/badge";
+import { Button } from "@/components/ui/button";
+import { CardContent, CardHeader } from "@/components/ui/card";
```
Comment (rithviknishad) [inline src/components/Patient/SampleDetails.tsx]: Let's not mix shadcn-ui's card with the other card. Switch to using shadcn's card.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/components/Patient/SampleDetails.tsx (hunk 4 above)
Comment (modamaan) [reply src/components/Patient/SampleDetails.tsx]: @rithviknishad shadcn Card added, old Card removed

**Comment by modamaan** [comment]
@rithviknishad

File: src/components/Patient/SampleDetails.tsx [hunk 5]
```diff
@@ -305,169 +306,230 @@ export const SampleDetails = ({ id }: DetailRoute) => {
...
+                  <div className="font-medium text-sm">
+                    {t("countries_travelled")}:{" "}
+                  </div>
+                  <Badge variant={"secondary"}>
+                    {" "}
+                    {sampleDetails.patient_travel_history}
+                  </Badge>
+                </div>
+              )}
+          </div>
```
[the remark this answers is not in the document]
Comment (rithviknishad) [reply src/components/Patient/SampleDetails.tsx:510]: @modamaan let's do this. code would look lot cleaner. Maybe InfoBadge can be kept as a local reusable component and not export it.

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**